Sprint_4

MRadyushin opened the pull request:
Финальный проект 4 спринта

MRadyushin committed:
first commit

MRadyushin committed:
Изменение Pom файла

MRadyushin committed:
Удаление файлов

MRadyushin committed:
Merge pull request #1 from MRadyushin/main

first commit

MRadyushin committed:
Создание класса AnnotationTest.
Класс содержит открытие и закрытие браузера.

MRadyushin committed:
Создание класса HomePage

MRadyushin committed:
Досоздание класса HomePage описывающий главную страницу

MRadyushin committed:
Досоздание класса HomePage описывающий главную страницу

MRadyushin committed:
Создание Теста по 1 -му заданию

MRadyushin committed:
Создание Теста по 1 -му заданию

MRadyushin committed:
Cоздание класса DesignPage описывающий страницу заказа

MRadyushin committed:
Cоздание 2 тестового задания

MRadyushin committed:
Правка на запуск через Chrome

egorvas commented on .idea/.gitignore:
```diff
@@ -0,0 +1,3 @@
+# Default ignored files
```
Рекомендую добавлять папку .idea в .gitignore всего репозитория. По сути файлы в папке .idea это твои локальный файлы и не нужны для работоспособности проекта, но при этом могут приватные данные.

egorvas commented on src/test/java/pagesData/DesignPage.java:
```diff
@@ -0,0 +1,155 @@
+    private final By popupWindow = By.xpath("//div[contains(@class, 'Order_ModalHeader')]");
+
+    // Локатор кнопка оформления заказа
+    private final By YesOrderButton = By.xpath("//*[@id=\'root\']/div/div[2]/div[5]/div[2]/button[2]");
```
Константы в java через SNAKE_UPPER_CASE
С большой буквы называются классы

egorvas commented on src/test/java/tests/RentScooterTests.java:
```diff
@@ -0,0 +1,74 @@
+    public void testMakeScooter() {
+        HomePage objHomePage = new HomePage(driver);
+        // Нажатие кнопки Заказать в заголовке
+        objHomePage.clickHomeButtonHeader();
```
По условиям задания "Заказ самоката. Весь флоу позитивного сценария. Обрати внимание, что есть две точки входа в сценарий: кнопка «Заказать» вверху страницы и внизу."
Ты же сделал два теста в которых у тебя нажимается на objHomePage.clickHomeButtonHeader()
Это нужно исправить.

egorvas commented on src/test/java/tests/AnnotationTests.java:
```diff
@@ -0,0 +1,39 @@
+        // Создаем драйвер под Google Chrome
+        driver = new ChromeDriver();
+        // Создаем драйвер под MozilaFirefox
+        //driver = new FirefoxDriver();
```
Вместо комментов лучше иметь if который будет тебе выбирать либо один driver, либо второй

MRadyushin committed:
Добавлены изменения:
1) в gitignore добавлено игнорирование соджержимого папки idea
2) Поправлено наименование локатора
3) Добавлен тест ,через вторую точку входа в Заказ
4) Добавлено условие выбора для зхапуска браузера
